sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java

```diff
@@ -246,31 +246,48 @@ private static void saveMetric(SensorContext sensorContext, InputFile inputFile,
     sensorContext.<Integer>newMeasure().forMetric(metric).on(inputFile).withValue(value).save();
   }
 
-  private void saveFailures(SensorContext sensorContext, Failure[] failures, TypeScriptRules typeScriptRules) {
+  private void saveIssues(SensorContext sensorContext, Issue[] issues, TypeScriptRules typeScriptRules) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. ([squid:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-244&issues=AWCije-U_Q-0dz4JAh0W&open=AWCije-U_Q-0dz4JAh0W)

[ISSUE_KEY]: AWCije-U_Q-0dz4JAh0W